libamqpprox/amqpprox_session.cpp

```diff
@@ -522,6 +523,30 @@ void Session::pause()
     }
 }
 
+void Session::unpause()
+{
+    if (d_sessionState.getPaused()) {
+        if (d_sessionState.getReadyToConnectOnUnpause()) {
+            BOOST_LOG_SCOPED_THREAD_ATTR(
+                "Vhost",
+                boost::log::attributes::constant<std::string>(
+                    d_sessionState.getVirtualHost()));
+            BOOST_LOG_SCOPED_THREAD_ATTR(
+                "ConnID",
+                boost::log::attributes::constant<uint64_t>(
+                    d_sessionState.id()));
+            LOG_INFO << "Session unpaused. Starting to acquire connection";
+
+            d_sessionState.setReadyToConnectOnUnpause(false);
+            d_sessionState.setPaused(false);
+            establishConnection();
+        }
+        else {
+            this->disconnect(true);
```

Comment:
Should we send the explicit connection Close method to client here, before closing sockets?

Reply:
Paused connections aren't paused at a frame boundary, so I don't think we can send any data down the socket before closing.